userspace/libsinsp/threadinfo.cpp

```diff
@@ -967,7 +969,46 @@ bool sinsp_threadinfo::is_health_probe()
 {
 	return (m_category == sinsp_threadinfo::CAT_HEALTHCHECK ||
 		m_category == sinsp_threadinfo::CAT_LIVENESS_PROBE ||
-		m_category == sinsp_threadinfo::CAT_READINESS_PROBE);
+	        m_category == sinsp_threadinfo::CAT_READINESS_PROBE);
+}
+
+string sinsp_threadinfo::get_path_for_dir_fd(int64_t dir_fd)
+{
+	sinsp_fdinfo_t* dir_fdinfo = get_fd(dir_fd);
+	if (!dir_fdinfo || dir_fdinfo->m_name.empty())
+	{
+#ifdef HAS_CAPTURE
+		// Sad day; we don't have the directory in the tinfo's fd cache.
+		// Must manually look it up so we can resolve filenames correctly.
+		char proc_path[PATH_MAX];
+		char dirfd_path[PATH_MAX];
+		int ret;
+		snprintf(proc_path,
+		         sizeof(proc_path),
+		         "/proc/%lld/fd/%lld",
+		         (long long)m_pid,
+		         (long long)dir_fd);
+
+		ret = readlink(proc_path, dirfd_path, sizeof(dirfd_path) - 1);
+		if (ret < 0)
+		{
+			g_logger.log("Unable to determine working directory for thread.",
```

Comment:
Not exactly working directory :) We're looking up the path to a specific fd, not to the thread's cwd